bluetti_mqtt/mqtt_client.py

```diff
@@ -113,9 +116,121 @@ async def _handle_command(self, mqtt_message: MQTTMessage):
+                                     unit_of_measurement='W',
+                                     device_class='power',
+                                     state_class='measurement',
+                                     force_update=True)
+                                     .encode(),
+                                 retain=True
+                                 )
+
+            await client.publish(f'homeassistant/sensor/{d.sn}_ac_output_power/config',
+                        payload=payload(
+                            id='ac_output_power',
+                            device=d,
+                            name='AC Output Power',
+                            unit_of_measurement='W',
+                            device_class='power',
+                            state_class='measurement',
+                            force_update=True)
+                            .encode(),
+                        retain=True
+                        )
+
+            await client.publish(f'homeassistant/sensor/{d.sn}_dc_output_power/config',
+                        payload=payload(
+                            id='dc_output_power',
+                            device=d,
+                            name='DC Output Power',
+                            unit_of_measurement='W',
+                            device_class='power',
+                            state_class='measurement',
+                            force_update=True)
+                            .encode(),
+                        retain=True
+                        )
+
+            await client.publish(f'homeassistant/sensor/{d.sn}_total_battery_percent/config',
+                        payload=payload(
+                            id='total_battery_percent',
+                            device=d,
+                            name='Total Battery Percent',
+                            unit_of_measurement='%',
+                            device_class='battery',
+                            state_class='measurement')
+                            .encode(),
+                        retain=True
+                        )
+
+            await client.publish(f'homeassistant/binary_sensor/{d.sn}_ac_output_on/config',
+                        payload=payload(
+                            id='ac_output_on',
+                            device=d,
+                            name='AC Output',
+                            device_class='power')
+                            .encode(),
+                        retain=True
+                        )
+
+            await client.publish(f'homeassistant/binary_sensor/{d.sn}_dc_output_on/config',
+                        payload=payload(
+                            id='dc_output_on',
+                            device=d,
+                            name='DC Output',
+                            device_class='power')
+                            .encode(),
+                        retain=True
+                        )
```

Comment:
Is it possible for these to be switches instead of just sensors? I don't use Home Assistant, personally, so I'm not sure whether that's possible or not.

Reply:
Yes, that should be possible. I will try that and report back.

Reply:
@lbossle, @warhammerkid, thanks so much for this work - I'm monitoring your progress and will be happy to test on my Airstream Bluetti/Home Assistant setup when you finish switches implementation (switches instead of sensors is exactly what I would need!!!)

Incidentally, looking forward to potentially contributing to a native integration as an adaptation if that becomes relevant.  MQTT is great, but it needs the intermediary device.  Also, with much more robust bluetooth integration in HA now rolled out, it could just be a native bluetooth integration built on this awesome library.

Reply:
Okay, I just replaced the binary sensors with switches. Please feel free to test.

Reply:
It works! I was confused at first because the binary sensors were still there, but then I remembered that I needed to clear them as a retained message to make them go away. Thanks for putting this together.